### PR TITLE
[PLAT-4933] Add insert command to RN CLI

### PR DESCRIPTION
**Pull request by bengourley**
Adds `bugsnag-react-native-cli insert` command.

Unit tests included for relevant parts. Interactive CLI aspects will be test via maze runner at a later point.

To test this out, use the following steps:

- `git fetch && git checkout feature/rn-cli-insert-cmd`
- `npm i`
- `npm run bootstrap`
- `npm run build`
- `cd packages/react-native-cli`
- `npm link` (this symlinks the local cli package to your global npm modules)

On a React Native project:

- check that the command is linked up correctly `bugsnag-react-native-cli --help`
- run `bugsnag-react-native-cli insert`
- verify that the Bugsnag initialisation has been correctly inserted into `index.js`, `MainApplication.java` and `AppDelegate.m`

### CODE REVIEW

**Review by nickdowell — APPROVED** [review]
Ran `bugsnag-react-native-cli insert` on a freshly created project.

LGTM 👍

File: packages/react-native-cli/src/lib/Insert.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+`import Bugsnag from "@bugsnag/react-native";
+Bugsnag.start();`
+
+const BUGSNAG_COCOA_IMPORT = '#import <Bugsnag/Bugsnag.h>'
+const BUGSNAG_COCOA_INIT = '[Bugsnag start];'
+const COCOA_APP_LAUNCH_REGEX = /(-\s*\(BOOL\)\s*application:\s*\(UIApplication\s\*\)\s*application\s+didFinishLaunchingWithOptions:\s*\(NSDictionary\s*\*\)launchOptions\s*\{\s*)\S/
+
+const BUGSNAG_JAVA_IMPORT = 'import com.bugsnag.android.Bugsnag;'
+const BUGSNAG_JAVA_INIT = 'Bugsnag.start();'
+const JAVA_APP_ON_CREATE_REGEX = /(public void onCreate\s*\(\)\s*\{[.\s]*super\.onCreate\([.\s]*\);(\s*))\S/
```
Comment (fractalwrench) [inline packages/react-native-cli/src/lib/Insert.ts:15]: There are a couple of scenarios I thought of which might fail if a user has customized their code a bit:

1. The user might decide to run some of their own code before calling `super.onCreate()`:

```
public void onCreate() {
    logSomething("onCreate() called");
    super.onCreate();
}
```

2. The `@Override` annotation is on the same line for some code styles:

```
@Override public void onCreate() {
```

My understanding is that it should be possible to support these use cases with the same approach - if not, they're probably not important enough to worry about.

**Review by fractalwrench — APPROVED** [review]

File: packages/react-native-cli/src/lib/Insert.ts (hunk 1 above)
Comment (bengourley) [reply packages/react-native-cli/src/lib/Insert.ts:15]: 1. should already be catered for with `[.\s]*` before `super\.onCreate` in the regex?
2. should also already be catered for? The regex doesn't care what comes before `public void onCreate`

perhaps worth adding a test for these cases to demonstrate

Comment (bengourley) [reply packages/react-native-cli/src/lib/Insert.ts]: Regex tweaked and tests added to for these cases.

**Pull request merged**